common/component/postgresql/databases/databases.go

```diff
@@ -0,0 +1,91 @@
+package databases
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pginterfaces "github.com/dapr/components-contrib/common/component/postgresql/interfaces"
+	"github.com/jackc/pgx/v5"
+)
+
+const (
+	createDatabaseTmpl = "CREATE DATABASE %s"
+	deleteDatabaseTmpl = "DROP DATABASE IF EXISTS %s WITH (FORCE)"
+	databaseExistsTmpl = "SELECT EXISTS (SELECT 1 FROM pg_database WHERE datname = $1)"
+	userExistsTmpl     = "SELECT CAST((SELECT 1 FROM pg_roles WHERE rolname = $1) AS BOOLEAN)"
+	createUserTmpl     = "CREATE USER %v WITH PASSWORD '%v'"
+	checkRoleTmpl      = "SELECT 1 FROM pg_roles WHERE rolname = $1 AND 'rds_iam' = rolname"
+	grantRoleTmpl      = "GRANT %v TO %v"
```

Comment:
Most of these queries use `%s` or other string formatting, so are potentially vulnerable to SQL injection attacks. Although this is a bit mitigated by the fact that the data comes from the component's metadata, so it's set by an admin, It would be great if we could try and use parametrized queries (e.g. queries that use `$1` et similia) instead. Otherwise, we may get this flagged in the next security audit!

If using parametrized queries is not possible (sometimes it's not for queries that change the system), we should sanitize the input to make sure it's limited to characters that can't cause SQL injections. Example, what we do for MySQL: https://github.dev/dapr/components-contrib/blob/49afc557e4d7b836e8077017c503248f8fb9d6aa/state/mysql/mysql.go#L825-L844

Reply:
addressed